mailet/standard/src/test/java/org/apache/james/transport/matchers/AbstractHasMailAttributeTest.java

```diff
@@ -30,42 +34,38 @@
 import org.apache.mailet.base.test.FakeMail;
 import org.apache.mailet.base.test.FakeMatcherConfig;
 import org.apache.mailet.base.test.MailUtil;
+import org.junit.jupiter.api.Test;
 
-import junit.framework.TestCase;
 
-public abstract class AbstractHasMailAttributeTest extends TestCase {
+public abstract class AbstractHasMailAttributeTest {
 
-    protected FakeMail mockedMail;
+    FakeMail mockedMail;
 
     protected Matcher matcher;
 
-    protected static final String MAIL_ATTRIBUTE_NAME = "org.apache.james.test.junit";
-
-    protected static final String MAIL_ATTRIBUTE_VALUE = "true";
+    static final String MAIL_ATTRIBUTE_NAME = "org.apache.james.test.junit";
 
-    protected String mailAttributeName = "org.apache.james";
+    static final String MAIL_ATTRIBUTE_VALUE = "true";
 
-    protected String mailAttributeValue = "false";
+    private String mailAttributeName = "org.apache.james";
 
-    public AbstractHasMailAttributeTest() {
-        super(null);
-    }
+    private String mailAttributeValue = "false";
 
-    protected void setMailAttributeName(String mailAttributeName) {
+    void setMailAttributeName(String mailAttributeName) {
         this.mailAttributeName = mailAttributeName;
     }
 
-    protected void setMailAttributeValue(String mailAttributeValue) {
+    void setMailAttributeValue(String mailAttributeValue) {
         this.mailAttributeValue = mailAttributeValue;
     }
 
-    protected void setupMockedMail() throws MessagingException {
+    void setupMockedMail() throws MessagingException {
         mockedMail = MailUtil.createMockMail2Recipients();
         mockedMail.setAttribute(mailAttributeName,
                 mailAttributeValue);
     }
 
-    protected void setupMatcher() throws MessagingException {
+    void setupMatcher() throws MessagingException {
```

Comment:
do you really want to prevent overriding and access on these methods?

Reply:
package private is overridable by child classes of the same package.

Reply:
yes I know, but why forcing the every child classes are in the same package?